subprojects/base-services/src/main/java/org/gradle/api/JavaVersion.java

```diff
@@ -101,6 +101,42 @@ public enum JavaVersion {
     @Incubating
     VERSION_20,
 
+    /**
+     * Java 21 major version.
+     * Not officially supported by Gradle. Use at your own risk.
+     *
+     * @since 7.6
+     */
+    @Incubating
+    VERSION_21,
+
+    /**
+     * Java 22 major version.
+     * Not officially supported by Gradle. Use at your own risk.
+     *
+     * @since 7.6
+     */
+    @Incubating
+    VERSION_22,
```

Comment:
Would it be worth `@Deprecating` the versions that don't yet exist, then un-deprecating them when support is added in Gradle?  It would draw people's attention to the lack of support in the code if they happen to use those versions, as the `@Incubating` can be kind of invisible.

Reply:
IMO it's not that big of a deal, usually they function just fine and people should use them. In the future we'll have no warnings at all if #22396 lands.